redbot/core/commands/commands.py

```diff
@@ -330,15 +330,26 @@ def parents(self) -> List["Group"]:
             if not change_permission_state:
                 ctx.permission_state = original_state
 
-    async def _verify_checks(self, ctx):
+    async def prepare(self, ctx):
+        ctx.command = self
+
         if not self.enabled:
             raise commands.DisabledCommand(f"{self.name} command is disabled")
 
-        if not (await self.can_run(ctx, change_permission_state=True)):
+        if not await self.can_run(ctx, change_permission_state=True):
             raise commands.CheckFailure(
                 f"The check functions for command {self.qualified_name} failed."
             )
 
+        if self.cooldown_after_parsing:
+            await self._parse_arguments(ctx)
+            self._prepare_cooldowns(ctx)
+        else:
+            self._prepare_cooldowns(ctx)
+            await self._parse_arguments(ctx)
+
```

Comment:
This doesn't call `max_concurrency`
```suggestion
        if self._max_concurrency is not None:
            await self._max_concurrency.acquire(ctx)
```

Reply:
Thanks!